src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -528,3 +570,77 @@ object IndexLogEntry {
 
   def schemaString(schema: StructType): String = schema.json
 }
+
+/**
+ * Provides functionality to generate unique file ids for files.
+ */
+class FileIdTracker {
```

Comment:
This class has many logics to validate and I think we need unit tests for this class. I will create an issue to follow up.

Reply:
Created #258